### PR TITLE
Improve rabbimq-server module start.

### PR DESCRIPTION
**Pull request by mrkz**
Check the status of the rabbitmq-server service
before trying to start it, providing more useful logs when
errors are found when using the --debug flag.

Signed-off-by: Simental Magana, Marcos marcos.simental.magana@intel.com

### CODE REVIEW

**Comment by albertomurillo** [comment]
lgtm +1

File: clearstack/modules/rabbitmq.py [hunk 1]
```diff
@@ -51,10 +52,37 @@ def user_exists(self, user):
...
+        LOG.debug("enabling rabbitmq-server service")
+        self.enable_server()
+        status = util.service_status("rabbitmq-server.service")
+        LOG.debug("rabbitmq-server status: " + status)
+        if status == "active":
+            LOG.debug("rabbitmq-server service is already running")
+            return
+        elif status == "inactive" or status == "unknown":
+            LOG.debug("starting rabbitmq-server service")
+            util.run_command("systemctl restart rabbitmq-server")
```
Comment (electrocucaracha) [inline clearstack/modules/rabbitmq.py:70]: It should be nice to have a util.restart_service wrapper method

File: clearstack/modules/rabbitmq.py [hunk 2]
```diff
@@ -51,10 +52,37 @@ def user_exists(self, user):
...
+        elif status == "failed":
+            # try to re-run the service, catch the error if fails again
+            try:
+                LOG.debug("rabbitmq-server failed to start previously. "
+                          "Trying to start service...")
+                util.run_command("systemctl restart rabbitmq-server")
```
Comment (electrocucaracha) [inline clearstack/modules/rabbitmq.py:76]: ditto line 70

File: clearstack/modules/rabbitmq.py [hunk 3]
```diff
@@ -51,10 +52,37 @@ def user_exists(self, user):
...
+            except:
+                LOG.debug("rabbitmq-server.service failed to start."
```
Comment (electrocucaracha) [inline clearstack/modules/rabbitmq.py:78]: Could you change the level of logging? IMHO this must be LOG.error or it should be raise an exception

File: clearstack/modules/rabbitmq.py [hunk 4]
```diff
@@ -51,10 +52,37 @@ def user_exists(self, user):
...
+                          "Try 'rm -rf /var/lib/rabbitmq/*' and re-run"
+                          "clearstack")
+                sys.exit(1)
+        else:
+            LOG.debug("Error: rabbitmq-server service status"
```
Comment (electrocucaracha) [inline clearstack/modules/rabbitmq.py:83]: ditto line 78

**Pull request merged**